iroh/src/magicsock/node_map/node_state.rs

```diff
@@ -699,6 +699,18 @@ impl NodeState {
         }
 
         let mut access = self.udp_paths.access_mut(now);
+
+        // Invalidate paths that are not in the new address set.
+        // If the remote node sends us new addresses, old ones are likely stale (e.g., after restart).
+        // This forces immediate re-evaluation and prevents preferring "known bad" outdated paths
+        // over "unknown potentially good" new paths.
```

Comment:
@Frando @flub can you look at this please, I believe the two of you looked at this before

Reply:
I'm also unsure, I'll have to dig in a bit more to find out if this could have unintended side effects (ie falsely invalidating good paths).